lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -241,6 +257,100 @@ def _run(self, params, concierge_params):
 
         return job_id
 
+    def _abort_child_jobs(self, child_job_ids):
+        """
+        Cancel a list of child jobs, and their child jobs
+        """
+        for child_job_id in child_job_ids:
+            try:
+                self.sdkmr.cancel_job(
+                    job_id=child_job_id,
+                    terminated_code=TerminatedCode.terminated_by_batch_abort.value,
+                )
+            except Exception as e:
+                # TODO Maybe add a retry here?
+                self.logger.error(f"Couldn't cancel child job {e}")
+
+    def _create_parent_job(self, wsid, meta):
+        """
+        This creates the parent job for all children to mark as their ancestor
+        :param params:
+        :return:
+        """
+        job_input = JobInput()
+        job_input.service_ver = "batch"
+        job_input.app_id = "batch"
+        job_input.method = "batch"
+        job_input.narrative_cell_info = Meta()
+
+        if meta:
+            job_input.narrative_cell_info.run_id = meta.get("run_id")
```

Comment:
Could rewrite this in a more concise way as

```py
for meta_attr in ['run_id', 'token_id', 'tag', 'cell_id', 'status']:
  job_input.narrative_cell_info[meta_attr] = meta.get(meta_attr)
```

Reply:
You are right! The MongoEngine object has a `__setitem__` so we can do it that way. Thanks, I changed it.